### PR TITLE
gatfruits and pea bullets (haters say nerf!)

### PR DESCRIPTION
**Pull request by capsaicinz**
## About The Pull Request

GATFRUITS! they're funny, pretty achievable, and pretty brutal in terms of damage. a whole .357 revolver, loaded up to the brim with 60-damage bullets. it's fun to stash your inventory with a few and go hunt prey, and i'm not removing that viability here- but it's going to be a bit less ruthless.

pea bullets! instead of shooting lead into your enemies, shoot a rather moderately damaging (15 brute, plus bullet embed) pea with also deals 5 toxin damage to your opponent! certainly nothing to chuckle about, and a rather unique spin you won't see in many other guns. i wanted to make sure the viability of this weapon wasn't damaged too hard, so here we are.

also, gives you a pretty botanical revolver instead of a whole-ass one.

-----

unknown
_still crits a completely unarmored monkey brain in a full magdump!_

-----

unknown
_look at those vague, unnamed vines growing on your shiny new gatfruit!_

## Why It's Good For The Game

in all seriousness, i think such an easily farmable gun (once achieved through the seed vault or xenobio) should be a little less than the traitor-specific gun that deals 60 brute damage. lowering it to 15 still keeps it as a viable weapon, and the toxin damage adds that little charm to it that makes it especially worth it, i think.

## Changelog
:cl:
add: A mutation in gatfruit seeds has led to a drastic alteration in the observable traits of the plant, which now fires hardened peas that deal less damage, but poison the target.
/:cl:

### CODE REVIEW

**Comment by capsaicinz** [comment]
we do a tad bit of committing..

**Comment by axietheaxolotl** [comment]
based master committer????

**Comment by EricZilla** [comment]
Peas please :)

**Comment by ANAME593** [comment]
the gatfruit is actually very rare you need somebody to actually go down to the seed vault or roll it from silver slimes i do believe this nerf will remove any sightings of gatfruit due to its rarity plus its no longer a very strong gun

**Review by Twaticus — APPROVED** [review]

**Comment by itseasytosee** [comment]
> the gatfruit is actually very rare you need somebody to actually go down to the seed vault or roll it from silver slimes i do believe this nerf will remove any sightings of gatfruit due to its rarity plus its no longer a very strong gun

Don't doubt the strength of embed, you can kill someone with 3 or less throwing stars if they move around at all.

**Comment by D4C-420** [comment]
does this deal bonus damage against zombies?

**Comment by VladinXXV** [comment]
This is cool but it's absolutely still a nerf. You're going from 2 bullets to a full mag to crit, it's about 3 times less effective. You didn't bother to change anything about how rare it is to obtain gatfruit, nor did you change the immensely long maturation time. Botany has infinitely many options that leave this in the dust. I would consider:
-making it easier to obtain
-lowering the maturation time
-bonus damage to zombies (it's too perfect to miss out on)
-make peas transfer plant reagents?

I'd also prefer going full plant for the sprites. I don't think slapping some vines on an otherwise normal looking gun really conveys anything.

**Comment by Cheshify** [comment]
Do we really care about gatfruit nerfs when they're the most unbalanced shit in the game? Oh boy botanists got a gatfruit seed, time for a FFA in the halls. This just reduces botany from being free antags to being free antags that can't instantly murder anyone they dislike, so that's pretty cool.

**Comment by Iamgoofball** [comment]
:-1:
the entire point of gatfruit is that it is a plant that grows .357 revolvers

there was a comment for like 8 years that was "todo: a plant that grows .357 revolvers"
if you want this, you need to make a plant that grows .357 revolvers as a mutation of this plant

File: code/modules/projectiles/projectile/bullets/revolver.dm
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:133]:
```suggestion
	if(. == BULLET_ACT_HIT && isliving(target))
```
If you only want the toxin damage to be dealt on a successful hit

**Review by MrMelbert — COMMENTED** [review]
I'd like to see bonus damage applied to zombies.
3 ricochets max instead of 2 would also be funny.

File: code/modules/projectiles/projectile/bullets/revolver.dm
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:135]: Better var names for this.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammunition/ballistic/revolver.dm
Comment (MrMelbert) [inline code/modules/projectiles/ammunition/ballistic/revolver.dm:71]: Also, effects after qdel icky:

- Drop `src` to the ground
- Put the new peas in hand + the to_hat
- THEN delete `src`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by VladinXXV** [comment]
> Do we really care about gatfruit nerfs when they're the most unbalanced shit in the game? Oh boy botanists got a gatfruit seed, time for a FFA in the halls. This just reduces botany from being free antags to being free antags that can't instantly murder anyone they dislike, so that's pretty cool.

I'm not saying nerfing is a bad thing. I think this is a cool idea that could use a few tweaks. Whether the nerf is needed or not, it's still a nerf and should be labeled as such. And it doesn't really reduce botany. This does not affect a competent botanists ability to throw a leaf at you and drop you in 4 seconds.

**Comment by VladinXXV** [comment]
Might as well rename gatfruit to peashooter pods or something while we're at it, since peas are not fruits.

File: code/modules/projectiles/projectile/bullets/revolver.dm
Comment (capsaicinz) [reply code/modules/projectiles/projectile/bullets/revolver.dm:136]: in the context of the bullet being blocked by something like a desword, or bouncing off armor?

**Comment by capsaicinz** [comment]
so, something i did in the last commit- gatfruits are now attainable through mutating normal peas. i think the damage nerf makes this acceptable, but i'd like to hear what you all think.

**Comment by Shadowflame909** [comment]
botany is back baby

**Comment by CMDR-Gungnir** [comment]
I think this is an awful change.  Botany can already make other things significantly better than this. If you want something like this, you'd be better off adding a new plant called Peashooter that mutates off of peas.

As it stands, this'll...probably pretty much never see any use. Let Gatfruit BE that rare, very slow to grow, but very good thing.

**Comment by jughu** [comment]
Anyone who says gatfruits are rare is delusional
You dont know what rare is
Sure it was rare when we didnt have lavaland
But now its as common as fuck as it spawna gauranteed in the seed vault ruin in lavaland

**Comment by capsaicinz** [comment]
alright, i'm probably going to leave it here. i want to let this simmer for awhile if it gets merged, and see what happens to the usage of gatfruit. i've undone the aiblity for it to be accessible through peas, but that very well might change if it's underused. crossed fingers, thanks for the feedback.

**Comment by Bawhoppen** [comment]
Why on earth did gatfruits get obtainable ingame through non-admin means

**Comment by Shadowflame909** [comment]
> Why on earth did gatfruits get obtainable ingame through non-admin means

Miner and xenobio incentive for departmental cooperation

**Comment by Bawhoppen** [comment]
departmental cooperation to get revolvers, huh...

Goof is right, let's just restore them to the fun classic they were, but non-obtainable for regular rounds. Peashooters can be added separately or something

**Comment by BlueMemesauce** [comment]
~~i'm the only one that knows you can get gatfruit from cargo exploration drones~~

**Comment by TheBonded** [comment]
I think a freezing variant would be cool

**Comment by VladinXXV** [comment]
> I think a freezing variant would be cool

If the peas were made to transfer reagents you could use frost oil. Flaming variant? Can probably get ClF3 from instability randomness or somethin.

File: code/modules/projectiles/projectile/bullets/revolver.dm
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:138]: Why are these not updating health

**Comment by capsaicinz** [comment]
i'm going to close this for now because i want to make it a little more intuitive than just less damage, and i have some plans for other funny botany things that seem wildly overpowered i  compared to what other departments have access to

**Comment by Toastgoats** [comment]
Do you still plan on working on this? I was thinking of doing something similar.

**Pull request closed without merging**
